llama-guard-safe-chatbot/Llama-Guard-3-Ensuring-Safe-Chatbot.ipynb

```diff
@@ -0,0 +1,410 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "3874bfd5-9699-4eaa-b952-81bf1ac61064",
+   "metadata": {},
+   "source": [
+    "# Llama Guard 3: Ensuring Safe User Interactions in Chatbots"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "bbcae649-22cd-4643-968b-727f3afd61da",
+   "metadata": {},
+   "source": [
+    "In today's digital landscape, chatbots play a vital role in facilitating user interactions across various platforms. However, with the increasing volume of user-generated messages, ensuring that these interactions remain safe and respectful is paramount. This is where content moderation becomes essential.\n",
+    "\n",
+    "[Meta's Llama Guard 3](https://llama.meta.com/docs/model-cards-and-prompt-formats/llama-guard-3/) is an advanced content moderation tool designed to assess user messages for harmful content. By analyzing incoming messages, Llama Guard 3 can effectively identify and classify harmful content across 14 categories, including hate speech, threats, and misinformation. This proactive approach allows chatbots to filter out harmful messages, ensuring that the language model (LLM) does not engage with or respond to inappropriate content.\n",
+    "\n",
+    "In this tutorial, we will demonstrate how utilizing Llama Guard 3 on Groq API effectively detects harmful content in user messages. By detecting harmful messages from users, you can ensure that your chatbot either does not respond or provides a generic or safe response that does not engage with the harmful content. We will provide examples of how the model identifies and classifies harmful messages across various categories, ensuring that inappropriate content is filtered out before the chatbot engages with users.\n",
+    "\n",
+    "By implementing Llama Guard 3 through the [Groq API](https://console.groq.com/playground), you can significantly enhance user safety, uphold community standards, and create a respectful online environment. This proactive approach not only protects users but also contributes to a more positive and engaging experience. Let's get started!"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d88eb5aa-5823-471d-9ee6-c57c89d2a81d",
+   "metadata": {},
+   "source": [
+    "### Setup"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "id": "55774e3d-d352-4226-ac1b-df581f7452aa",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from groq import Groq\n",
+    "import pandas as pd\n",
+    "import random\n",
+    "\n",
+    "client = Groq()"
```

Comment:
are you going to pass in the API key as an env variable here and mention in the notebook or just leave it like this?

Reply:
Yeah good point, will be clearer to pass in the key explicitly with the env variable